tox.ini

```diff
@@ -57,3 +57,16 @@ deps =
     -r{toxinidir}/requirements.txt
     -r{toxinidir}/dev_requirements.txt
     -e.
+
+[testenv:integration-spark-session]
+basepython = python3
+commands = /bin/bash -c '{envpython} -m pytest -v tests/specs/spark-session.dbtspec'
```

Comment:
This fails because the HIVE format is not found. Even though it prints:

```
Spark will be initialized with options:
  spark.app.name: dbt-spark
  spark.default.parallelism: 1
  spark.dynamicAllocation.enabled: false
  spark.executor.cores: 1
  spark.executor.instances: 1
  spark.io.compression.codec: lz4
  spark.rdd.compress: false
  spark.shuffle.compress: false
  spark.sql.catalogImplementation: hive      <----
  spark.sql.shuffle.partitions: 1
```

I don't really understand how this `dbtspec` works. It feels like it starts another python session, which then does **not** have the hive format once it `getOrCreate` the Spark session.

@jtcohen6 : Could you provide some guidance on the `dbtspec` stuff?

Reply:
Ok, if I hard code the catalog implementation setting to hive inside the `Cursor` I do not get this error anymore. That confirms my hunch that another python session is created.

Also, I can not jump into a debug trace to start debugging, which confirms the hunch too - and is annoying to be honest

Reply:
The whole `dbtspec` situation is something we're planning to move away from, in favor of a new testing framework that's integrated across `dbt-core` and adapter plugins (rather than the 2+ different frameworks we support today).

For more on why `dbtspec` (`pytest-dbt-adapter`) is lackluster at supporting adapter developers, you can read: https://github.com/dbt-labs/dbt-adapter-tests/issues/25. The particular complaints are that it's hard to see what's going on behind the scenes (too much magic), and hard to do any sort of interactive debugging, as you mention.